spring-cloud-kubernetes-commons/src/main/java/org/springframework/cloud/kubernetes/commons/config/PropertySourceUtils.java

```diff
@@ -80,12 +80,16 @@ private PropertySourceUtils() {
 			yamlFactory.setDocumentMatchers(properties -> {
 				if (environment != null) {
 					String profiles = null;
-					if (properties.containsKey(SPRING_CONFIG_ACTIVATE_ON_PROFILE)) {
```

Comment:
this is a common anti-pattern: `contains`, followed by `get`; which performs _two_ hash look-ups. Instead, issue a `get` followed by a check against `null`.